data/templates/postmigrate.sh

```diff
@@ -5,38 +5,48 @@ cd ${TEMP_DIR}
 PREUPGRADE_LOG=/var/log/preupgrade.log
 touch ${PREUPGRADE_LOG}
 TAR_BALL=preupgrade.tar.gz
+echo "INFO: prepare tarball ${TAR_BALL}." >> ${PREUPGRADE_LOG}
 echo "{tar_ball}" > data
 base64 --decode data > ${TAR_BALL}
-tar -xzvf ${TAR_BALL}
+echo "INFO: unpack tarball ${TAR_BALL}." >> ${PREUPGRADE_LOG}
+tar --selinux -xzvf ${TAR_BALL}
+# create symlinks
 ln -s {RESULT_NAME}/cleanconf cleanconf
 ln -s {RESULT_NAME}/dirtyconf dirtyconf
+ln -s {RESULT_NAME}/kickstart kickstart
+ln -s {RESULT_NAME}/postmigrate.d postmigrate.d
+ln -s {RESULT_NAME}/noauto_postupgrade.d noauto_postupgrade.d
+
 ls -laR >> ${PREUPGRADE_LOG}
-cd {RESULT_NAME}
-PWD=${pwd}
-cd cleanconf
+cd {RESULT_NAME}/cleanconf
+echo "INFO: Restore configuration files from ${PWD} directory." >> ${PREUPGRADE_LOG}
 for file in $(find . -type f)
 do
     ABS_PATH=${file:1}
-    SAVE_PATH="${ABS_PATH}.rpmsave"
-    DIFF_FILENAME="$(basename ${file}).diff"
-    echo "Comparing file ${file} against ${SAVE_PATH}" >> ${PREUPGRADE_LOG}
-    echo "Diff output is stored in ${TEMP_DIR}/${DIFF_FILENAME}" >> ${PREUPGRADE_LOG}
-    [[ -f ${ABS_PATH} ]] && \
+    SAVE_PATH="${ABS_PATH}.preupg_save"
+    if [[ -f "${ABS_PATH}" ]]
+    then
+        echo "INFO: Back up new configuration file to '${SAVE_PATH}'." >> ${PREUPGRADE_LOG}
```

Comment:
I wouldn't call the conf file that you are backing up as "new". When I looked at the log I was confused a bit. And present continuous tense may be more suitable too.
I suggest something like:
"INFO: Backing up the clean system configuration file '${ABS_PATH}' to '${SAVE_PATH}'."

An then for the restoration:
"INFO: Restoring configuration file of the migrated system '${ABS_PATH}'."